client/wfnews-war/src/main/angular/src/app/components/draggable-panel/draggable-panel.component.ts

```diff
@@ -462,58 +500,84 @@ return 'Unknown';
               .getEvacOrdersByEventNumber(
                 this.identifyItem.properties.EVENT_NUMBER,
                 {
-                  returnGeometry: false,
-                  returnCentroid: false,
-                  returnExtent: true,
+                  returnGeometry: true,
                 },
               )
               .toPromise()
               .then((response) => {
-                if (response?.extent) {
-                  viewer.map.fitBounds([
-                    [response.extent.ymin, response.extent.xmin],
-                    [response.extent.ymax, response.extent.xmax]
-                ]);
-                }
+                  if (response?.features?.length > 0 && response?.features[0].geometry?.rings?.length > 0){
+                    const polygonData = this.extractPolygonData(response.features[0].geometry.rings);
+                    if (polygonData.length) {
+                      this.fixPolygonToMap(polygonData);
+                    }
+                  }
               });
           } else if (layerId.includes('area-restrictions')) {
             this.agolService
               .getAreaRestrictions(
                 `NAME='${this.identifyItem.properties.NAME}'`,
                 null,
                 {
-                  returnGeometry: false,
-                  returnCentroid: false,
-                  returnExtent: true,
+                  returnGeometry: true,
+                },
+              )
+              .toPromise()
+              .then((response) => {
+                if (response?.features?.length > 0 && response?.features[0].geometry?.rings?.length > 0){
+                  const polygonData = this.extractPolygonData(response.features[0].geometry.rings);
+                  if (polygonData.length) {
+                    this.fixPolygonToMap(polygonData);
+                  }                
+                }
+              });
+          } else if (layerId.includes('danger-rating')) {
+            this.agolService
+              .getDangerRatings(
+                `PROT_DR_SYSID ='${this.identifyItem.properties.PROT_DR_SYSID}'`,
+                null,
+                {
+                  returnGeometry: true,
                 },
               )
               .toPromise()
               .then((response) => {
-                viewer.map.fitBounds([
-                    [response.extent.ymin, response.extent.xmin],
-                    [response.extent.ymax, response.extent.xmax]
-                ]);
+                if (response?.features?.length > 0 && response?.features[0].geometry?.rings?.length > 0){
+                  const polygonData = this.extractPolygonData(response.features[0].geometry.rings);
+                  if (polygonData.length) {
+                    this.fixPolygonToMap(polygonData);
+                  }                
+                }
               });
           }
+        } else if ( // local authorities
+          layerId === 'abms-regional-districts' ||
```

Comment:
Optional as always, but can be simplified

```
['abms-regional-districts', 'clab-indian-reserves', 'abms-municipalities'].includes(layerId)
```

Reply:
sure